internal/pkg/flink/components/table_view.go

```diff
@@ -73,25 +117,65 @@ func (t *TableView) getSelectedRowIdx() int {
 	return rowIdx
 }
 
-func (t *TableView) RenderTable(tableTitle string, statementResults *types.MaterializedStatementResults, isAutoRefreshRunning bool) {
+func (t *TableView) RenderTable(tableTitle string, statementResults *types.MaterializedStatementResults, isAutoRefreshRunning bool, lastRefreshTimestamp *time.Time, fetchState types.FetchState) {
 	t.tableLock.Lock()
 	defer t.tableLock.Unlock()
 
-	t.rootLayout = createTableView(t.table, NewShortcuts(t.getTableShortcuts(statementResults, isAutoRefreshRunning)))
+	t.infoBar.SetLastRefreshTimestamp(lastRefreshTimestamp)
+	t.infoBar.SetFetchState(fetchState)
+	t.createTableView(NewShortcuts(t.getTableShortcuts(statementResults, isAutoRefreshRunning)))
+	t.setTableAndColumnWidths(statementResults)
+
 	t.table.SetTitle(tableTitle)
 	t.renderData(statementResults)
 	t.selectLastRow(!isAutoRefreshRunning)
 }
 
-func (t *TableView) renderData(statementResults *types.MaterializedStatementResults) {
-	t.table.Clear()
+func (t *TableView) createTableView(shortcuts *tview.TextView) {
+	interactiveOutput := InteractiveOutput(t.table, t.infoBar.GetView(), shortcuts)
+	t.rootLayout = RootLayout(interactiveOutput)
+}
 
+func (t *TableView) setTableAndColumnWidths(statementResults *types.MaterializedStatementResults) {
 	_, _, tableWidth, _ := t.table.GetInnerRect()
 	t.tableWidth = tableWidth
-	columnWidths := statementResults.GetMaxWidthPerColumn()
-	truncatedColumnWidths := results.GetTruncatedColumnWidths(columnWidths, tableWidth)
+	t.columnWidths = statementResults.GetMaxWidthPerColumn()
+}
+
+func (t *TableView) tableAfterDrawHandler() func(screen tcell.Screen, x int, y int, width int, height int) (int, int, int, int) {
```

Comment:
Isn't this just a  `TableDrawFunction` or `TableDrawFunctionWrapper`? Handler is also alright, but it wasn't so intuitive to understand what's that for

Reply:
Hmm it's a callback that's invoked after the table has been drawn, so I think the name is fitting, no?

Reply:
Will leave it as it is for now